azure-iot-device/samples/advanced-hub-scenarios/provision_symmetric_key_and_send_telemetry.py

```diff
@@ -0,0 +1,69 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import asyncio
+from azure.iot.device.aio import ProvisioningDeviceClient
+import os
+from azure.iot.device.aio import IoTHubDeviceClient
+from azure.iot.device import Message
+import uuid
+
+
+messages_to_send = 10
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+
+async def main():
+    async def register_device():
+        provisioning_device_client = ProvisioningDeviceClient.create_from_symmetric_key(
+            provisioning_host=provisioning_host,
+            registration_id=registration_id,
+            id_scope=id_scope,
+            symmetric_key=symmetric_key,
+        )
+
+        return await provisioning_device_client.register()
+
+    results = await asyncio.gather(register_device())
+    registration_result = results[0]
+    print("The complete registration result is")
+    print(registration_result.registration_state)
```

Comment:
This is a strange style.  Why not just skip the register_device coroutine and the gather call.  Neither of these seem to do anything?

e.g.
```
async def main():
    provisioning_device_client = ProvisioningDeviceClient.create_from_symmetric_key(
        provisioning_host=provisioning_host,
        registration_id=registration_id,
        id_scope=id_scope,
        symmetric_key=symmetric_key,
    )
    registration_result = await provisioning_device_client.register()

    print("The complete registration result is")
    print(registration_result.registration_state)
```

Reply:
Yeah, I agree. Gather only makes sense when there are multiple operations.

---
In reply to: [354614793](https://github.com/Azure/azure-iot-sdk-python/pull/407#discussion_r354614793) [](ancestors = 354614793)